gui/tools/Helper.h

```diff
@@ -0,0 +1,227 @@
+  static std::vector<std::string> ExtractAllKRTDFilePath(const char* globalPath, const char* framelist)
+  {
+    std::vector<std::string> pathList;
+
+    // Open file which contains the list of all file
+    std::ifstream container(framelist);
+    if (!container.is_open())
+    {
+      std::cerr << "Unable to open : " << framelist << std::endl;
+      return pathList;
+    }
+
+    std::string path;
+    while (!container.eof())
+    {
+      std::getline(container, path);
+      // only get the file name, not the whole path
+      std::vector <std::string> elems;
+      help::SplitString(path, ' ', elems);
+      // check if there are an empty line
+      if( elems.empty() )
+      {
+        continue;
+      }
+      std::vector <std::string> filename;
+      help::SplitString(elems[elems.size() - 1], '/', filename);
+
+      // Create the real data path to access depth map file
+      std::vector <std::string> elemsWithoutExtension;
+      help::SplitString(filename[filename.size() - 1], '.', elemsWithoutExtension);
+      pathList.push_back(std::string(globalPath) + "/" + elemsWithoutExtension[0] + ".krtd");
+    }
+
+    return pathList;
+  }
+
+  //----------------------------------------------------------------------------
+  // Description
+  // Read krtd file and create K and RT matrix
+  static bool ReadKrtdFile(std::string filename, vtkMatrix3x3* matrixK,
+                           vtkMatrix4x4* matrixRT)
+  {
+    // Open the file
+    std::ifstream file(filename.c_str());
+    if (!file.is_open())
+    {
+      std::cerr << "Unable to open krtd file : " << filename << std::endl;
+      return false;
+    }
+
+    std::string line;
+
+    // Get matrix K
+    for (int i = 0; i < 3; i++)
+    {
+      getline(file, line);
+      std::istringstream iss(line);
+
+      for (int j = 0; j < 3; j++)
+      {
+        double value;
+        iss >> value;
+        matrixK->SetElement(i, j, value);
+      }
+    }
+
+    getline(file, line);
+
+    // Get matrix R
+    for (int i = 0; i < 3; i++)
+    {
+      getline(file, line);
+      std::istringstream iss(line);
+
+      for (int j = 0; j < 3; j++)
+      {
+        double value;
+        iss >> value;
+        matrixRT->SetElement(i, j, value);
+      }
+    }
+
+    getline(file, line);
+
+    // Get matrix T
+    getline(file, line);
+    std::istringstream iss(line);
+    for (int i = 0; i < 3; i++)
+    {
+      double value;
+      iss >> value;
+      matrixRT->SetElement(i, 3, value);
+    }
+
+    // Finalize matrix RT
+    for (int j = 0; j < 4; j++)
+    {
+      matrixRT->SetElement(3, j, 0);
+    }
+    matrixRT->SetElement(3, 3, 1);
+
+    return true;
+  }
+
+
+  //----------------------------------------------------------------------------
+  // Description
+  // Compute median of a vector
+  template <typename T>
+  static void ComputeMedian(std::vector<T> vector, double& median)
+  {
+    std::sort(vector.begin(), vector.end());
```

Comment:
It's more efficient to compute the median using std::nth_element.  See for example 

http://stackoverflow.com/questions/1719070/what-is-the-right-approach-when-using-stl-container-for-median-calculation 

Technically the median requires averaging the two center elements for an even length vector.  However, in practice we often don't worry about that and just take closest index to the center.  e.g. `vector.size() / 2`

Ideally this helper function should move out of the GUI code, but that's not critical.